### PR TITLE
Features/issue 21 flask

### PR DESCRIPTION
**Pull request by crankycoder**
Port of the taar-api to use Flask instead of Django

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 
 setup(
     name='taar_api',
     version='0.2dev',
     description='This is https://github.com/mozilla/taar_api',
     author='Mozilla Foundation',
     author_email='',
-    url='https://github.com/mozilla/taar_api'
+    url='https://github.com/mozilla/taar_api',
+    packages=find_packages(exclude=['tests', 'tests/*']),
```
Comment (crankycoder) [inline setup.py]: Maybe this is why we have to set PYTHONPATH now

File: tox.ini [hunk 2]
```diff
@@ -17,7 +16,9 @@ sitepackages=True
 deps =
     pytest
     pytest-cov
-    pytest-django
```
Comment (crankycoder) [inline tox.ini]: pytest-django modifies the PYTHONPATH in https://github.com/pytest-dev/pytest-django/blob/a67ab3bd0eda8a2b76b4c5954b04482b4f27e20a/pytest_django/plugin.py#L123

This means when we switch over to Flask, the tox.ini needs the PYTHONPATH explicitly setup to point at /app

File: setup.py (hunk 1 above)
Comment (crankycoder) [reply setup.py]: Nope.  See: https://github.com/mozilla/taar-api/pull/37/files#r175086624

File: bin/run [hunk 3]
```diff
@@ -28,16 +28,15 @@ wait_for() {
...
 # Only wait for backend services in development
 # http://stackoverflow.com/a/13864829
-[ ! -z ${DEVELOPMENT+check} ] && wait_for db 5432 && wait_for redis 6379
+# [ ! -z ${DEVELOPMENT+check} ] && wait_for db 5432 && wait_for redis 6379
 
 case $1 in
   web)
-    newrelic-admin run-python manage.py migrate --noinput
-    exec newrelic-admin run-program gunicorn taar_api.wsgi:application -b 0.0.0.0:${PORT} --workers 4 --access-logfile -
+    #exec newrelic-admin run-program gunicorn taar_api.app:app -b 0.0.0.0:${PORT} --workers 4 --access-logfile -
```
Comment (birdsarah) [inline bin/run]: lets take all these commented out lines before we merge. also in dockerfile.

Comment (crankycoder) [reply bin/run]: Fixed now.  I've also parameterized the port and host for the web-dev target by adding an extra wsgi wrapper, so bin/run is very minimally altered now.

**Comment by crankycoder** [comment]
Ok, I think this is all good now.

I've tried splitting the changes up so that it's easier to digest.

Roughly - to get an overview of the changes with respect to the 'docker' changes and not the 'python' changes - you can look at :  

`git diff master Dockerfile docker-compose.yml bin/`

The rest of the file changes are 'regular' django->flask changes.

File: tox.ini [hunk 4]
```diff
@@ -1,5 +1,8 @@
 [tox]
 envlist = tests, flake8
+toxworkdir=/tmp/tox
```
Comment (crankycoder) [inline tox.ini]: Running tests through tox required extra configuration in tox.ini as we are now running with uid 10001 as per the [dockerflow](https://github.com/mozilla-services/Dockerflow) spec and write permission has been tightened up in the Dockerfile.

File: taar_api/app.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    locale = request.args.get('locale', None)
+    if locale is not None:
+        extra_data['locale'] = locale
+
+    platform = request.args.get('platform', None)
+    if platform is not None:
+        extra_data['platform'] = platform
+
+    if PROXY_MANAGER.getResource() is None:
+        ctx = default_context()
```
Comment (birdsarah) [inline taar_api/app.py]: This whole block of code is not under test coverage.

File: bin/run [hunk 6]
```diff
@@ -28,16 +28,14 @@ wait_for() {
...
 # Only wait for backend services in development
 # http://stackoverflow.com/a/13864829
-[ ! -z ${DEVELOPMENT+check} ] && wait_for db 5432 && wait_for redis 6379
+# [ ! -z ${DEVELOPMENT+check} ] && wait_for db 5432 && wait_for redis 6379
 
 case $1 in
   web)
-    newrelic-admin run-python manage.py migrate --noinput
-    exec newrelic-admin run-program gunicorn taar_api.wsgi:application -b 0.0.0.0:${PORT} --workers 4 --access-logfile -
+    exec newrelic-admin run-program gunicorn taar_api.app:app -b 0.0.0.0:${PORT} --workers 4 --access-logfile -
```
Comment (birdsarah) [inline bin/run]: I don't love the disconnectedness between the port exposed in the Dockerfile, the port forwarded in the docker-compose, and the port specified in /bin/run.

I know this isn't new to your PR. Just an observation.

**Pull request merged**